### PR TITLE
community: add new parameter default_headers

### PR DESCRIPTION
**Pull request by lu-ohai**
Thank you for contributing to LangChain!

- [x] **PR title**: "package: description"
  - "community: 1. add new parameter `default_headers` for oci model deployments and oci chat model deployments. 2. updated k parameter in OCIModelDeploymentLLM class."

- [x] **PR message**:
    - **Description:** 1. add new parameters `default_headers` for oci model deployments and oci chat model deployments. 2. updated k parameter in OCIModelDeploymentLLM class.

- [x] **Add tests and docs**:
  1. unit tests
  2. notebook

### CODE REVIEW

File: libs/community/langchain_community/llms/oci_data_science_model_deployment_endpoint.py [hunk 1]
```diff
@@ -472,6 +482,25 @@ def _identifying_params(self) -> Dict[str, Any]:
...
+            is_async (bool, optional): Indicates if the request is asynchronous.
+                Defaults to `False`.
+            body (optional): The request body to be included in the headers if
+                the request is asynchronous.
+
+        Returns:
+            Dict: A dictionary containing the appropriate headers for the request.
+        """
+        return {
+            "route": DEFAULT_INFERENCE_ENDPOINT,
```
Comment (efriis) [inline libs/community/langchain_community/llms/oci_data_science_model_deployment_endpoint.py:500]: should this be included as a header?

**Review by efriis — COMMENTED** [review]
one question /suggestion to remove the route from headers. otherwise looks good. 

Also, if you're interested in maintaining this integration without us in the loop, we'd love to get an integration package out! Future PRs against langchain would just be {docs updates, as well as registering your package in `libs/packages.yml`, deprecating this community integration in favor of your integration package}

Here's the guide, and if you have questions, feel free to leave them in the comments on those pages so others can see them! https://python.langchain.com/docs/contributing/how_to/integrations/

There's some folks within oracle working on this that may be interested in collaborating

File: libs/community/langchain_community/llms/oci_data_science_model_deployment_endpoint.py (hunk 1 above)
Comment (mrDzurb) [reply libs/community/langchain_community/llms/oci_data_science_model_deployment_endpoint.py:500]: Hi @efriis, thank you for your comment! In the current implementation of OCI Model Deployment, the `route` parameter needs to be passed through the header. For text and chat models, this parameter will be hardcoded by default. However, users can modify the value of the `route` parameter if needed by utilizing the `default_header` attribute.

**Comment by mrDzurb** [comment]
> one question /suggestion to remove the route from headers. otherwise looks good.
> 
> Also, if you're interested in maintaining this integration without us in the loop, we'd love to get an integration package out! Future PRs against langchain would just be {docs updates, as well as registering your package in `libs/packages.yml`, deprecating this community integration in favor of your integration package}
> 
> Here's the guide, and if you have questions, feel free to leave them in the comments on those pages so others can see them! https://python.langchain.com/docs/contributing/how_to/integrations/
> 
> There's some folks within oracle working on this that may be interested in collaborating

Hi @efriis, moving to the integration package is already on our roadmap! Thanks!

File: libs/community/langchain_community/llms/oci_data_science_model_deployment_endpoint.py (hunk 1 above)
Comment (efriis) [reply libs/community/langchain_community/llms/oci_data_science_model_deployment_endpoint.py:500]: sounds good

**Pull request merged**